tools: add a simple FAQ for DM to cover `invalid connection` and `driver: bad connection` (#1299)
pingcap/docs: v3.0/faq/data-migration.md

---

> title: Data Migration FAQ

Reviewer: ```suggestion
title: Data Migration FAQ
summary: Learn about two questions (FAQs) relating to Data Migration.
```

category: FAQ
---

# Data Migration FAQ

> 

Reviewer: ```suggestion

This is a simple FAQ page for DM, including:

+ How to handle the `invalid connection` error
+ How to handle the `driver: bad connection` error

```

## What can I do when a synchronization task is interrupted with an `invalid connection` error returned ?

> `invalid connection` error usually indicates connection error between DM and downstream TiDB database (Network failure, TiDB restart, Tikv busy and so on) and some data of current request have been sent to TiDB.

Reviewer: ```suggestion
`invalid connection` error usually indicates that errors have occurred in the connection between DM and the downstream TiDB database (such as network failure, TiDB restart, TiKV busy and so on) and that part of the data for the current request has been sent to TiDB.
```

> DM will concurrently replicate data downstream in synchronize tasks. Due to this feature, an interrupted task may contain several errors. ( You could check current error by using `query-status` or `query-error`. )

Reviewer: ```suggestion
DM concurrently replicates data downstream in its synchronization tasks. Due to this feature, several errors might occur when a task is interrupted. You can check these errors through `query-status` or `query-error`.
```

> - If there is only `invalid connection` error during incremental replication, DM will retry automatically.

Reviewer: ```suggestion
- If there is only `invalid connection` error during the incremental replication, DM retries automatically.
```

> - If DM doesn't or fail to retry automatically because of version problems, you can use `stop-task` to stop the task and then use`start-task` to restart the task.

Reviewer: ```suggestion
If DM does not or fails to retry automatically because of version problems, use `stop-task` to stop the task and then use`start-task` to restart the task.
```

> ## What can I do when a synchronization task is interrupted with an `driver: bad connection` error ?

Reviewer: ```suggestion
## What can I do when a synchronization task is interrupted with the `driver: bad connection` error returned?
```

> `driver: bad connection` usually indicates connection error between DM and upstream TiDB database (Network failure, TiDB restart and so on) and no data of current request are sent to TiDB temporarily.

Reviewer: ```suggestion
`driver: bad connection` error usually indicates that errors have occurred in the connection between DM and the upstream TiDB database (such as network failure, TiDB restart and so on) and that data of the current request has not been sent to TiDB at that moment.
```

> In the current version, when facing this type of error, you should use `stop-task` to stop the task and then use `start-task` to restart the task. DM will improve its automatic retry function for this type of error later.

Reviewer: ```suggestion
When facing this type of error in the current version, use `stop-task` to stop the task and then use `start-task` to restart the task. DM will be improved on its automatic retry mechanism to cope with this type of error in the future.
```

---
title: Data Migration FAQ
summary: Learn about two questions (FAQs) relating to Data Migration.
category: FAQ
---

# Data Migration FAQ

> This is a simple FAQ page for DM, including:

Reviewer: ```suggestion
This is a simple FAQ page for Data Migration (DM), including:
```

+ How to handle the `invalid connection` error
+ How to handle the `driver: bad connection` error

> ## What can I do when a synchronization task is interrupted with an `invalid connection` error returned ?

Reviewer: ```suggestion
## What can I do when a replication task is interrupted with an `invalid connection` error returned?
```

> `invalid connection` error usually indicates that errors have occurred in the connection between DM and the downstream TiDB database (such as network failure, TiDB restart, TiKV busy and so on) and that part of the data for the current request has been sent to TiDB.

Reviewer: ```suggestion
`invalid connection` error usually indicates that errors have occurred in the connection between DM and the downstream TiDB database (such as network failure, TiDB restart, TiKV busy and so on) and that a part of the data for the current request has been sent to TiDB.
```

> DM concurrently replicates data downstream in its synchronization tasks. Due to this feature, several errors might occur when a task is interrupted. You can check these errors through `query-status` or `query-error`.

Reviewer: ```suggestion
DM concurrently replicates data downstream in its replication tasks. Due to this feature, several errors might occur when a task is interrupted. You can check these errors using `query-status` or `query-error`.
```

> - If there is only `invalid connection` error during the incremental replication, DM retries automatically.

Reviewer: ```suggestion
- If there is only the `invalid connection` error returned during the incremental replication, DM retries automatically.
```

If DM does not or fails to retry automatically because of version problems, use `stop-task` to stop the task and then use`start-task` to restart the task.

> ## What can I do when a synchronization task is interrupted with the `driver: bad connection` error returned?

Reviewer: ```suggestion
## What can I do when a replication task is interrupted with the `driver: bad connection` error returned?
```

> `driver: bad connection` error usually indicates that errors have occurred in the connection between DM and the upstream TiDB database (such as network failure, TiDB restart and so on) and that data of the current request has not been sent to TiDB at that moment.

Reviewer: ```suggestion
`driver: bad connection` error usually indicates that errors have occurred in the connection between DM and the upstream TiDB database (such as network failure, TiDB restart and so on) and that the data of the current request has not been sent to TiDB at that moment.
```
